src/hooks/useExpertModeQueryParamReader.ts

```diff
@@ -0,0 +1,31 @@
+import { parse } from 'qs'
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { RouteComponentProps } from 'react-router-dom'
+import { AppDispatch } from 'state'
+import { updateUserExpertMode } from '../state/user/actions'
+
+export default function ExpertModeQueryParamReader({ location: { search } }: RouteComponentProps): null {
+  return useExpertModeQueryParamReader(search)
+}
+
+function useExpertModeQueryParamReader(search: string): null {
+  const dispatch = useDispatch<AppDispatch>()
+
+  useEffect(() => {
+    if (!search) return
+
+    const parsed = parse(search, {
```

Comment:
we have something called useParsedQuery... for this

Reply:
nice, thanks.. didn't know that existed